lib/portlayer/exec/commit.go

```diff
@@ -138,45 +135,33 @@ func Commit(ctx context.Context, sess *session.Session, h *Handle, waitTime *int
 			// NOTE: this inline refresh can be removed when switching away from guestinfo where we have non-persistence issues
 			// when updating ExtraConfig via the API with a powered on VM - we therefore have to be absolutely certain about the
 			// power state to decide if we can continue without nilifying extraconfig
+			s := h.Spec.Spec()
 
-			for s := h.Spec.Spec(); ; refresh, s = true, h.Spec.Spec() {
-				// FIXME!!! this is a temporary hack until the concurrent modification retry logic is in place
-				if refresh {
-					base, err := h.updates(ctx)
-					if err == nil {
-						h.Runtime = base.Runtime
-						h.Config = base.Config
-					}
-				}
+			// poor man's test and set
+			s.ChangeVersion = h.Config.ChangeVersion
```

Comment:
You are right, task in progress will fail but I'm not sure whether we should treat it like a conflict error. Right now we get a TaskInProgress error first and then retry in the task package and get ConccurrentAccess error and return that and retry the whole operation in persona.